components/NavigationBar/examples/NavigationBarIconsExample.m

```diff
@@ -77,10 +77,13 @@ - (void)viewDidLoad {
   self.navigationItem.rightBarButtonItem = trailingButtonItem;
   self.navigationItem.backBarButtonItem = backButtonItem;
 
-  if ([self.view respondsToSelector:@selector(safeAreaLayoutGuide)]) {
```

Comment:
Can we identify the top constraint in an if statement?
That the execution path is the same, just with a slightly different parameter.
NSLayoutYAxisAnchor *topLayoutAnchor = self.topLayoutGuide.topAnchor;
// Determine best way to identify if we have a safeAreaLayoutGuide and add as the if clause.
if ([UIView respondsTo:safeAreaLayoutGuide]) {
  topLayoutAnchor = self.safeLayoutBuide.topAnchor.
}
[topLayoutAnchor constraintEqualToAnchor:self.navigationBar.topAnchor].active = YES;

Reply:
We can't use `-constraintEqualToAnchor:` until we drop iOS 8 support.  Do you want the following?
1. A path for iOS 8 only using NSLayoutConstraint ...withItem:
2. A second path for iOS 9+ that uses safeAreaInsets if available? We still need the macros to protect older Xcode/base SDK versions from seeing the new symbol.

This would mean we can migrate straight to `-constraintEqualToAnchor` by simply deleting the iOS 8 path later.

Reply:
iOS 8!!! :shakes_fist: